src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -406,7 +414,8 @@ PosixEventEngine::CreateEndpointFromUnconnectedFdInternal(
 #if GRPC_PLATFORM_SUPPORTS_POSIX_POLLING
   PosixError err;
   int connect_errno;
-  PosixEventPoller* poller = poller_manager_.Poller();
+  CHECK(poller_manager_.has_value());
```

Comment:
This runtime check needed to be added 7 times, which feels like a code smell, and is error-prone if we forget to add it to any function (now or later). I know you're interested in removing PollerManager. If we're going to take this approach of having an optional PollerManager that is checked every time it's used, we need a short finite timeline before these checks are removed. Please add a TODO and explain what's needed before we can remove the need for this to be an optional type / remove the checks / delete the PollerManager.